apps/frontend/src/app/3_organisms/ZeroLocForm/components/OpenCreditLine.tsx

```diff
@@ -0,0 +1,264 @@
+  const [collateralAmount, setCollateralAmount] = useState('0');
+  const [debtAmount, setDebtAmount] = useState('0');
+  const [debtToken, setDebtToken] = useState<SupportedTokens>(BORROW_ASSETS[0]);
+
+  const debtSize = useMemo(() => Number(debtAmount || 0), [debtAmount]);
+  const collateralSize = useMemo(
+    () => Number(collateralAmount || 0),
+    [collateralAmount],
+  );
+
+  const { value: maxCollateralWeiAmount } = useAssetBalance(
+    SupportedTokens.rbtc,
+  );
+
+  const originationFee = useMemo(
+    () => getOriginationFeeAmount(debtSize),
+    [debtSize],
+  );
+
+  const debtWithFees = useMemo(
+    () =>
+      debtSize > 0
+        ? getTotalDebtAmount(debtSize, borrowingRate, liquidationReserve)
+        : 0,
+    [debtSize, borrowingRate, liquidationReserve],
+  );
+
+  const rbtcGasPrice = useGasPrice();
+
+  const maxRbtcWeiBalance = useMemo(
+    () =>
+      BigNumber.from(maxCollateralWeiAmount)
+        .sub(composeGas(rbtcGasPrice || '0', MAX_GAS_LIMIT))
+        .toString(),
+    [maxCollateralWeiAmount, rbtcGasPrice],
+  );
+
+  const minCollateralAmount = useMemo(() => {
+    return (
+      (MIN_DEBT_SIZE / rbtcPrice) *
+      (isRecoveryMode ? CRITICAL_COLLATERAL_RATIO : MINIMUM_COLLATERAL_RATIO)
+    );
+  }, [isRecoveryMode, rbtcPrice]);
+
+  const maxCollateralAmount = useMemo(
+    () => Number(fromWei(maxRbtcWeiBalance)),
+    [maxRbtcWeiBalance],
+  );
+
+  const maxDebtAmount = useMemo(() => {
+    const rbtcBalance = Number(fromWei(maxRbtcWeiBalance));
+    const collateral = Number(collateralAmount);
+
+    let amount =
+      (rbtcBalance * rbtcPrice) /
+      (isRecoveryMode ? CRITICAL_COLLATERAL_RATIO : MINIMUM_COLLATERAL_RATIO);
+
+    if (collateral > 0) {
+      amount =
+        (collateral * rbtcPrice) /
+        (isRecoveryMode ? CRITICAL_COLLATERAL_RATIO : MINIMUM_COLLATERAL_RATIO);
+    }
+
+    const originationFee = getOriginationFeeAmount(amount, borrowingRate);
+
+    return Math.max(amount - originationFee - liquidationReserve, 0);
+  }, [
+    borrowingRate,
+    collateralAmount,
+    isRecoveryMode,
+    liquidationReserve,
+    maxRbtcWeiBalance,
+    rbtcPrice,
+  ]);
+
+  const ratio = useMemo(() => {
+    if ([collateralAmount, debtAmount, rbtcPrice].some(v => !v)) {
+      return 0;
+    }
+    return (
+      ((Number(collateralAmount) * Number(rbtcPrice)) / debtWithFees) * 100 || 0
+    );
+  }, [collateralAmount, debtAmount, debtWithFees, rbtcPrice]);
+
+  const { t } = useTranslation();
+
+  const liquidationPrice = useMemo(
+    () => MINIMUM_COLLATERAL_RATIO * (debtSize / Number(collateralAmount)),
+    [debtSize, collateralAmount],
+  );
+  const liquidationPriceInRecoveryMode = useMemo(
+    () => CRITICAL_COLLATERAL_RATIO * (debtSize / Number(collateralAmount)),
+    [collateralAmount, debtSize],
+  );
+
+  const errors = useMemo(() => {
+    if (!fieldsTouched) {
+      return [];
+    }
+    const list = checkForSystemErrors(ratio, tcr);
```

Comment:
```suggestion
    const errors = checkForSystemErrors(ratio, tcr);
```
Rename it in `AdjustCreditLine` as well.